project-koku/masu
# Load earlier months of data when adding new providers.

**Overview**
It is desired that multiple months of data is initially pulled into masu/koku when a new provider account is added.  This change will now download and process the current and previous month by default when a new account is added.  New environment variables are added so we can adjust these in production (`INITIAL_INGEST_OVERRIDE` and `INITIAL_INGEST_NUM_MONTHS`).

**Implementation Highlights**
* The downloader was changed accept an integer parameter to specify how many months of data to download.  (I.e. 2 => current month and previous month)
* Downloader now returns the billing start date as part of the report dictionary.
* The processor needs the billing start date so that it can calculate the summary table data for prior months.
* A new field was added to the provider model to keep track of the initial ingest of report data.

**Testing**
1. Create an S3 bucket with a reporting policy and 3 months of data loaded.  Verify that the initial download/processing includes 2 months of data (current month and previous month)
2. Update the current month's data and re-ingest.  Verify that only the current month is downloaded/processed.
3. Set the INITIAL_INGEST_OVERRIDE env to True and INITIAL_INGEST_NUM_MONTHS to 3.  Verify that the current-2 month is downloaded and processed.  current-1 and current month is not re-processed since they have not had any updates.

Test Results:
[masu_multimonth_initial_ingest_ut.txt](https://github.com/project-koku/masu/files/2390266/masu_multimonth_initial_ingest_ut.txt)
